### ISSUE TITLE
Unexpected problem using SSL

### ISSUE THREAD

**Issue by donmattes** [ckrey/MQTT-Client-Framework#301]
I have been using the MQTT-Client-Framework very successfully but have run into a snag and perhaps I am missing something. I am trying to add the use of SSL.  I have tried several scenarios.  The certificate I am using has worked successfully from other clients (like MQTTfx) to connect to the broker. (Using Rabbit with the MQTT add on)  Below is a snippet of the code.  I never get a return or execution of the connection handler as an error or success.  

NSString *certificate = [[NSBundle mainBundle] pathForResource:@"cacert" ofType:@"der"];
_mySession.securityPolicy = [MQTTSSLSecurityPolicy         policyWithPinningMode:MQTTSSLPinningModeCertificate] _mySession.securityPolicy.pinnedCertificates = @[ [NSData dataWithContentsOfFile:certificate] ] _mySession.securityPolicy.allowInvalidCertificates = YES;
        
[_mySession connectToHost:_brokerHost port:_brokerPort usingSSL:_useSSL connectHandler:^(NSError *error) {
        if (error) {
            NSLog(@"\n<<< Error in Connection Handler: %@\n", error.localizedDescription);
            _isBrokerConnected = false;
        } else {
            NSLog(@"\n<<< Success in Connection Handler\n");
	    _isBrokerConnected = true;
        }
}];

The following messages are logged in less than .5 seconds with debugging on but the 
connection handler is never fired.
		
[MQTTSSLSecurityPolicyTransport] open
[MQTTCFSocketEncoder] setState 0/0
[MQTTCFSocketDecoder] NSStreamEventOpenCompleted
[MQTTCFSocketEncoder] NSStreamEventOpenCompleted
[MQTTCFSocketEncoder] NSStreamEventHasSpaceAvailable
[MQTTCFSocketEncoder] setState 0/2
[MQTTCFSocketEncoder] NSStreamEventHasSpaceAvailable

**Comment by donmattes** [comment]
Update: After some additional work I was able to get to a point where an error status was triggered and the OS Error raised helped to resolve the issue.  In the end I got the correct certificate and the logic as outlined above worked as expected using the pinned certificate.  I did end up adding some additional _mysession.securitypolicy.[flags] based on information in the .h file to disable some other activities in the current case.  So things are running.  I do think that this may still be a bug in that it got to a location where the connect did not error out OR succeed and the completion handler was not executed.  I suspect that it is someplace in the code that is negotiating the SSL with the pinned certificate.
